### ISSUE TITLE
V.0.74-V.0.76: Message "Cannot connect to server" when trying to open a radio station (stream)

### ISSUE THREAD

**Issue by just-Nob** [segler-alex/RadioDroid#529]
As described in the issues title.
With the previous versions, eventing worked well.
Currently switched back to version 0.73.

**Comment by morckx** [comment]
@segler-alex : Could this also be related to the switch to uuids?

**Comment by just-Nob** [comment]
Unfortunately, this issue is still present in version 0.7.5.
Examples of an radio stations not able to play in V. 0.7.5:
* progrock.com
* N-Joy Radio
* Fritz (rbb)
* delta radio Der beste RockPop reloaded
* ...as well as some others.
Switched back to V. 0.7.3 one mir time.

**Comment by just-Nob** [comment]
Unfortunately, the problem still stays on in version 0.76.

Die to the fact that RadioDroid works fine on an older Android tablet (ASUS ZenPad 8.0 Tablet Model P022, with Intel processor inside, running Android 5.02), I guess I should give some more detailed info about _my_ Android mobile device:
* BQ Aquaris X Pro
* Android Oreo (8.1.0)
* Safety patch level from 01. July 2019
* Kennel version 3.18.71
* Qualcomm® Snapdragon™ 626-Prozessor (arm64 architecture, octacore)
* root

Anyway, due to the fact that on the one hand V.0.73 had no problems with the named streams and on the other hand V.0.74 to V.0.76 (latest one up to now) are still able to play some other streams, I presume that the playing capabilities of a given internet radio stream depends on the stream' s format.

**Comment by just-Nob** [comment]
Have checked it out on my girlfriend's mobile phone as well, which is very similar to mine (Moto G5 plus /XT1685 with Android Oreo 8.1.0, Qualcomm Snapdragon 625 arm64 processor), but without root.

The webstream radio station "progrock.com" works well, but the following stations (only examples!) don't work with RadioDroid's newer versions (German message: "Es könnte keine Verbindung zur Radiostation hergestellt werden."):
* delta radio Der beste RockPop reloaded
* NDR N-Joy
* Fritz RBB
* ...for sure many others...

**Comment by morckx** [comment]
I'm quite sure that this is related to deprecation of station ids in favour of uuids in the [radiobrowser-api](https://github.com/segler-alex/radiobrowser-api) and the required change in RadioDroid [here](https://github.com/segler-alex/RadioDroid/commit/d0030c655a7ffef45055ef503ee9f325b773f561).

While we should indeed provide an automatic conversion of old favourites and history lists, a simple workaround should be to delete the non-working stations from your favourites, search them and add them again. Please let us know if it worked.

> delta radio Der beste RockPop reloaded
NDR N-Joy
Fritz RBB
...for sure many others...

They all work for me when I play them from the search.

**Comment by just-Nob** [comment]
Hi @morckx ,

thanks for the feedback.
I wasn't aware of that uuid change.

Indeed, using the search instead of favorites / history list for playing a radio station / stream works well.

Unfortunately, it's not possible to remove favorites in RadioDroid's current version (V.0.7.6), so the workaround doesn't work for now.
Because that is another bug, I'm going to create a new issue.

**Comment by just-Nob** [comment]
Should this issue be closed out stay open as a reminder, due to the fact that you intend to implement the automatic uuid conversion?

**Comment by morckx** [comment]
Hi @just-Nob ,

> Indeed, using the search instead of favorites / history list for playing a radio station / stream works well.
>

Thanks for the confirmation! 

> Unfortunately, it's not possible to remove favorites in RadioDroid's current version (V.0.7.6), so the workaround doesn't work for now.
> Because that is another bug, I'm going to create a new issue.

When in list view, you should be able to just swipe stations to the left or to the right in order to delete them from the list. Doesn't that work?

Only when in tile view, currently only reordering (long-click) is possible.

> Should this issue be closed out stay open as a reminder, due to the fact that you intend to implement the automatic uuid conversion?

Please leave it open until we have the automatic conversion. I think that could be also a good milestone for a new stable release, before the full-screen player is coming.

**Comment by just-Nob** [comment]
Hi @morckx ,

unfortunately, I lost my "debugging environment" (i.e. the favorites' settings which caused the problems) due to an automated "Titanium Backup"-Backup 😉 that I hadn't had in mind...

In fact, there was only one entry (the old "ProgRock.com"-favorite) that I wasn't able to remove by tapping the "favorite"-star (I wasn't aware of the "sweetie away" method at this time, so I haven't tried that): RadioDroid told me that it removed the favorite, but it still stayed in the list, no matter how often I tried.
So I tried successfully the "remove all favorites" method, but was too lazy to immediately recover the old data by using Titanium Backup.
Last night, the automatic backup replaced my "old ", original data including all the favorites I had... 😔

Anyway, the "swipe to delete favorite" option works well for every new added favorite/station.
